### PR TITLE
refactor: deprecate Element APIs in ConfirmDialog

### PR DESCRIPTION
**Pull request by mstahv**
## Description

Element API should not be exposed to the component level.

## Type of change

Refactor

### CODE REVIEW

**Comment by mstahv** [comment]
Don't know how much these methods are used (hope not much). If somewhere else but in our own ITs, should be deprecated first 🤷‍♂️

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (knoobie) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:64]:
```suggestion
        dialog.setText(new Paragraph("Do you want to publish this post?"));
```

Stop abusing the label ;)

**Review by web-padawan — CHANGES_REQUESTED** [review]
Thanks for the PR. Unfortunately we can't remove existing methods that accept `Element` without deprecating them first. So let's mark them as `@Deprecated` now and then remove in the next major.

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (mstahv) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:64]: *It wasn't me it was the other guy* 🤪

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (tomivirkki) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:64]: The build is failing because there's no import for `Paragraph`

**Review by tomivirkki — CHANGES_REQUESTED** [review]
The title and the description of this PR need updates

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (knoobie) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:64]:
```suggestion
        dialog.setText(new Span("Do you want to publish this post?"));
```

if nobody wants to check out the branch again :) (Span was already importet)

Comment (mstahv) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:64]: 💪

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java
Comment (tomivirkki) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java:378]: Rather use the `Component` APIs in `SlotUtils.setSlot` also
```suggestion
        SlotUtils.setSlot(this, "reject-button", component);
```

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (mstahv) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:67]: Now see what you did @knoobie, not this is stuck in the system 🤪

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java
Comment (tomivirkki) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java]: Note that there are more than just this once instance. Didn't want to repeat myself so added the comment here only.

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (knoobie) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:67]: GitHub didn't let me suggest changes to the imports because those weren't changed 🙈

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java
Comment (mstahv) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java]: 🤦 Now I wonder if I checked that "allow maintainers to edit" checkbox ;-)

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (tomivirkki) [inline vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:9]: Aaand now the build doesn't pass because you removed the VaadinIcon import

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java
Comment (tomivirkki) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow/src/main/java/com/vaadin/flow/component/confirmdialog/ConfirmDialog.java]: That would help. I tried to push earlier but couldn't. But since you work at Vaadin, you should not use your own forks but work with a `flow-components` clone directly. You can request to be added to the `components-contributors` team for write access.

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (mstahv) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:9]: 🤬 The bloody project don't build properly anymore because time has passed (and probably my snapshots our out of sync). 

I have  a suggestion for you:

 * stop nitting, at least for PRs made by non-team members and PRs that have been hanging in the queue for more than a day or two
 * if there are tiny details you want to change in this kind of PRs, just change them. There should be no need for anybody to review those if you work with the codebase daily. 

This saves your time, my time and @knoobie's time.

Comment (knoobie) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:9]: I understand both sides here 😬 Just an idea from an outsider, that I've stolen from the spring team: they polish external contribution after the merge, making the process for both parties way more attractive and faster 😇

Comment (tomivirkki) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:9]: As I already said in https://github.com/vaadin/flow-components/pull/5495#discussion_r1345854188, I tried to push the fixes to your branch (regarding the "nitty" things) but since you don't allow maintainers to edit, I have no other choice than asking you to fix them.

**Review by tomivirkki — APPROVED** [review]

File: vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java
Comment (mstahv) [reply vaadin-confirm-dialog-flow-parent/vaadin-confirm-dialog-flow-integration-tests/src/main/java/com/vaadin/flow/component/confirmdialog/tests/FeaturesDiy.java:9]: That is a good one @knoobie 👍 Based on what I discussed with mr Spring Jürgen, they also don't have that strict core review processes, but do it when they see it appropriate. And then Jürgen checks all through before the release still. I think a lot more maintenance tasks could be done if we'd rationalize our processes on trivial tasks.

And BTW. My checkbox is on, we learn by doing ;-)

**Review by web-padawan — APPROVED** [review]

**Pull request merged**
